### DISCUSSION TITLE
How to add Console-address : port in kubernetes Yaml files for minio

### DISCUSSION THREAD

**Discussion by NisharunnisaShaik** [minio/minio#12684] [Q&A]
Hi Team  , We recently updated our Minio which is deployed on Kubernetes (EKS-V.1.18) with Latest Image RELEASE.2020-05-01T22-19-14Z
Since then we are unable to access our minio instance , below is the logs
**WARNING: Console endpoint is listening on a dynamic port (41417), please use --console-address ":PORT" to choose a static port.**

We are unable to roll back to older version seems its not working as expected too !

Kindly help me how to resolve this issue & make our minio up & running

**Comment by ravindk89** [comment]
> Latest Image RELEASE.2020-05-01T22-19-14Z

This is a very old image. The warning you are seeing would only come up if you upgraded past the 2021-06-17 release. I'll assume that is what you meant here.

How are you deploying onto Kubernetes? The MinIO Operator does not yet provide the necessary fields for setting a static embedded Console port. We also require v1.19 for the latest stable Operator. 

Can you also be specific as to what you mean about access? Is it just the embedded browser that you cannot access, or has your entire MinIO tenant gone down? If so, without logs we cannot really provide any troubleshooting. 

With regards to the rollback - we have and will continue to advise against making blind upgrades, especially if you are skipping many weeks or months of updates in between.

Adding some details from Slack for anyone else poking at this issue. 

With embedded Console, MinIO auto-redirects access to HOSTNAME:9000 -> the Console port. For containerized environments, this redirect might not work as the connecting client might not have access to that internal IP/port. This may break certain workflows that relied on the legacy browser.

The answer for non-Operator installations is to pass the `--console-port :PORT` argument in your YAML and ensure your ingress/load balancer/container config exposes those ports. You can add these arguments to your existing `args` array using the standard Kubernetes API. The Kubernetes docs has [examples](https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#running-a-command-in-a-shell)

For MinIO Tenants deployed via the Operator, the Operator can deploy a standalone MinIO Console you can use instead of the embedded console. This standalone Console is deployed by default if you created the tenant using the MinIO Kubernetes Plugin or the Operator Console UI. Future operator updates may further optimize deployments given the embedded Console functionality.

Regardless of environment, if you cannot migrate off the legacy embedded browser, do not upgrade past RELEASE.2021.06.17. And generally perform all upgrades in a testing or QA environment before applying them to any critical infrastructure.

**Comment by harshavardhana** [comment]
> Hi Team , We recently updated our Minio which is deployed on Kubernetes (EKS-V.1.18) with Latest Image RELEASE.2020-05-01T22-19-14Z
> Since then we are unable to access our minio instance , below is the logs
> **WARNING: Console endpoint is listening on a dynamic port (41417), please use --console-address ":PORT" to choose a static port.**
> 
> We are unable to roll back to older version seems its not working as expected too !
> 
> Kindly help me how to resolve this issue & make our minio up & running

MinIO is always up and running - are you trying to access browser?

**Reply by NisharunnisaShaik** [reply]
Hi @harshavardhana Yes i am trying to access minio via browser which is throwing error

**Reply by harshavardhana** [reply]
That is not an error just a warning @NisharunnisaShaik

**Reply by NisharunnisaShaik** [reply]
Hi @harshavardhana  I understand that the error i am getting is just a Warning but my point is we are unable to access our minio server via deployed Service_URL , in logs i can see the Endpoint & console of minio are running at someip:9000 but unable to expose that minio via Loadbalancer

**Reply by harshavardhana** [reply]
@NisharunnisaShaik you need to read https://github.com/minio/minio#test-minio-connectivity and things to consider.

**Reply by NisharunnisaShaik** [reply]
Hi @harshavardhana Earlier we do not have Env Variables like Minio_Redirect_URL , still we were able to access our minio instance on webbrowser. But from few days (after redeploying minio) we are unable to access Minio on Webserver. For Testing i have added MINIO_BROWSER_REDIRECT_URL in our deployment file with the our loadbalancer URL still unable to access minio endpoint from browser. The logs still say minio is running on someip:9000

**Reply by harshavardhana** [reply]
> Hi @harshavardhana Earlier we do not have Env Variables like Minio_Redirect_URL , still we were able to access our minio instance on webbrowser. But from few days (after redeploying minio) we are unable to access Minio on Webserver. For Testing i have added MINIO_BROWSER_REDIRECT_URL in our deployment file with the our loadbalancer URL still unable to access minio endpoint from browser. The logs still say minio is running on someip:9000

We have changed things you should read the release notes @NisharunnisaShaik

**Comment by zhang10609** [comment]
@NisharunnisaShaik I had resolved the issue and some comments had been listed in the below link pls refer it.
https://github.com/laradock/laradock/pull/3020
